query.go

```diff
@@ -1008,25 +996,44 @@ func (s *ChainService) GetBlock(blockHash chainhash.Hash,
 				// know about it and we declare it sane. We can
 				// kill the query and pass the response back to
 				// the caller.
-				foundBlock = block
-				close(quit)
+				foundBlock <- block
+				return true, true
 			default:
+				return false, false
 			}
+
 		},
-		options...,
+	}
+
+	msgs := []*query.Query{q}
+	errChan := s.queryAccess.Query(
+		msgs, translateQueryOptions(options...)...,
 	)
-	if foundBlock == nil {
-		return nil, fmt.Errorf("Couldn't retrieve block %s from "+
-			"network", blockHash)
+
+	var block *btcutil.Block
+
+Loop:
+	// Wait for the block, or an error to occur.
+	for {
```

Comment:
Why do we need this loop? The work manager can only deliver `nil` or `ErrWorkManagerShuttingDown`.